src/bootloader.rs

```diff
@@ -0,0 +1,500 @@
+use core::ops::Range;
+use std::{
+    collections::HashMap,
+    ffi::OsStr,
+    fs::{create_dir_all, write},
+    os::unix::ffi::OsStrExt,
+    path::Path,
+    str::from_utf8,
+};
+
+use rustix::{buffer::spare_capacity, io::read};
+
+use anyhow::{bail, ensure, Result};
+
+use crate::{
+    fsverity::FsVerityHashValue,
+    repository::Repository,
+    tree::{Directory, FileSystem, ImageError, Inode, LeafContent, RegularFile},
+};
+
+fn split_cmdline(cmdline: &str) -> impl Iterator<Item = &str> {
+    let mut in_quotes = false;
+
+    cmdline.split(move |c: char| {
```

Comment:
Huh I hadn't seen the model of passing a stateful closure to split, but it does look pretty clean in this case.

Reply:
Ya!  This was a really nice discovery for me too, thanks for noticing.  I may have been a tiny bit proud of this code after having written it. :)